src/Nether.Analytics/OutputFormatters/IHeaderAwareOutputFormatter.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Nether.Analytics
+{
+    /// <summary>
+    /// The interface enables the output formatter to be aware of the header. 
+    /// It also provides certain methods that enable features like serializing the 
+    /// message with the header for certain scenarios (i.e. CSV).
+    /// </summary>
+    public interface IHeaderAwareOutputFormatter : IOutputFormatter
```

Comment:
I'm a bit reluctant to do this - if I understand your comment correctly, that would couple the output manager directly to output formatter, which is exactly what I wanted to avoid in the first place. If I put it back into the OutputManager, then it has to understand the format directly (i.e. it has to know that it's two lines of CSV strings that can be joined with a NewLine). The idea is, OutputManagers don't ever need to call this unless they need a concatenated header and payload, but implementing it in this place reduces coupling and doesn't add any overhead. If a Formatter doesn't support it, they simply don't implement this.